includes/core/classes/class-settings.php

```diff
@@ -778,4 +779,25 @@ public function select_menu( $submenu ): string {
 		return (string) $submenu;
 	}
 
+	/**
+	 *  Display notice if users can't register
+	 *
+	 * @return void
+	 */
+	public function check_users_can_register() {
+		if ( filter_var( get_option( 'users_can_register' ), FILTER_VALIDATE_BOOLEAN ) ) {
+			return;
+		}
+		?>
+			<div class="notice notice-warning is-dismissible">
```

Comment:
Avoid markup in classes. Let's put this in a template and utilize `Utility::render_template` here. There's examples of format throughout the plugin you can follow. Put template in `gatherpress/includes/templates/admin/`.